catalog/async/catalog-async-postingestplugin/src/main/java/org/codice/ddf/catalog/async/processingplugin/ProcessingPostIngestPlugin.java

```diff
@@ -0,0 +1,206 @@
+                LOGGER.debug(
+                        "{} request property was not a boolean. Clearing the property and returning true. PostProcessingPlugins will not be run as an infinite loop may occur.",
+                        POST_PROCESS_COMPLETE);
+                properties.remove(POST_PROCESS_COMPLETE);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static Map<String, Serializable> putPostProcessCompleteFlagAndGet(
+            Map<String, Serializable> properties) {
+        Map<String, Serializable> newProperties = new HashMap<>(properties);
+        newProperties.put(POST_PROCESS_COMPLETE, true);
+        return newProperties;
+    }
+
+    private ProcessRequest<ProcessCreateItem> createCreateRequest(CreateResponse createResponse) {
+        List<ProcessCreateItem> processCreateItems;
+
+        processCreateItems = createResponse.getCreatedMetacards()
+                .stream()
+                .map(metacard -> new ProcessCreateItemImpl(getProcessResource(metacard),
+                        metacard,
+                        false))
+                .collect(Collectors.toList());
+
+        return new ProcessRequestImpl(processCreateItems,
+                putPostProcessCompleteFlagAndGet(createResponse.getProperties()));
+    }
+
+    private ProcessRequest<ProcessUpdateItem> createUpdateRequest(UpdateResponse updateResponse) {
+        List<Update> updates = updateResponse.getUpdatedMetacards();
+        List<ProcessUpdateItem> processUpdateItems = new ArrayList<>();
+
+        for (Update update : updates) {
+            Metacard oldCard = update.getOldMetacard();
+            Metacard newCard = update.getNewMetacard();
+            ProcessUpdateItem processItem = new ProcessUpdateItemImpl(getProcessResource(newCard),
+                    newCard,
+                    oldCard,
+                    false);
+            processUpdateItems.add(processItem);
+        }
+
+        return new ProcessRequestImpl(processUpdateItems,
+                putPostProcessCompleteFlagAndGet(updateResponse.getProperties()));
+    }
+
+    private ProcessRequest<ProcessDeleteItem> createDeleteRequest(DeleteResponse deleteResponse) {
+        List<ProcessDeleteItem> processDeleteItems = deleteResponse.getDeletedMetacards()
+                .stream()
+                .map(ProcessDeleteItemImpl::new)
+                .collect(Collectors.toList());
+
+        return new ProcessRequestImpl(processDeleteItems,
+                putPostProcessCompleteFlagAndGet(deleteResponse.getProperties()));
+    }
+
+    private ProcessResource getProcessResource(Metacard metacard) {
+        LOGGER.trace(
+                "Attempting to retrieve process resource metacard with id \"{}\" and sourceId \"{}\".",
+                metacard.getId(),
+                metacard.getSourceId());
+
+        ResourceRequest request = new ResourceRequestById(metacard.getId());
+
+        return Security.runAsAdmin(() -> {
```

Comment:
Do you even need this? You should have the subject attached to the request that is attached to the response. I don't think you should be elevating here.